noaa_test.go

```diff
@@ -105,3 +107,40 @@ func TestChicagoHourly(t *testing.T) {
 		t.Error("expected at least one period")
 	}
 }
+
+func TestSetClient(t *testing.T) {
+	// Intentionally create a client with an absurd timeout value.
+	client := &http.Client{
+		Timeout: time.Millisecond,
+	}
+
+	// Don't set client to ensure this still works as normal.
+	_, err := noaa.HourlyForecast("41.837", "-87.685")
+	if err != nil {
+		t.Errorf("should have successfully returned a result instead of this error: %s", err)
+	}
+
+	// Set the client to test this feature.
+	orig := noaa.SetClient(client)
+	if orig != nil && orig != http.DefaultClient {
```

Comment:
Client should never be nil given changes suggested to `SetClient(...)` so this could also get cleaned up.

Reply:
Ack.